File: app/src/main/java/com/rd/siri/model/AppState.kt
package com.rd.siri.model

sealed class VoiceState {
    object Idle : VoiceState()
    data class Loading(val message: String = "模型加载中…") : VoiceState()
    object Listening : VoiceState()
    object Recognizing : VoiceState()
    object Thinking : VoiceState()
    object Speaking : VoiceState()
    data class Error(val message: String) : VoiceState()
}

data class AppState(
    val voiceState: VoiceState = VoiceState.Loading(),
    val enginesReady: Boolean = false,
    val partialAsrText: String = "",
    val finalAsrText: String = "",
    val assistantReply: String = "",
    val hasConfig: Boolean = false,
    val wakeWordEnabled: Boolean = false,
    val connectionTestResult: String? = null
)

File: ios/SiriApp/Models/AppState.swift
//
//  AppState.swift
//  SiriApp
//
//  Ported from Android: AppState.kt
//

import Foundation

enum VoiceState: Equatable {
    case idle
    case loading(String)
    case listening
    case recognizing
    case thinking
    case speaking
    case error(String)

    var message: String {
        switch self {
        case .idle: return ""
        case .loading(let msg): return msg
        case .listening: return "正在聆听…"
        case .recognizing: return "识别中…"
        case .thinking: return "思考中…"
        case .speaking: return "播报中…"
        case .error(let msg): return msg
        }
    }
}

struct AppState {
    var voiceState: VoiceState = .loading("模型加载中…")
    var enginesReady: Bool = false
    var partialAsrText: String = ""
    var finalAsrText: String = ""
    var assistantReply: String = ""
    var hasConfig: Bool = false
    var ttsEnabled: Bool = true
}
